core/src/main/java/com/microsoft/applicationinsights/internal/channel/samplingV2/FixedRateSamplingTelemetryProcessor.java

```diff
@@ -0,0 +1,212 @@
+
+    private Set<Class> excludedTypes;
+
+    private Set<Class> includedTypes;
+
+
+    /// All sampling percentage must be in a ratio of 100/N where N is a whole number (2, 3, 4, …). E.g. 50 for 1/2 or 33.33 for 1/3.
+    /// Failure to follow this pattern can result in unexpected / incorrect computation of values in the portal.
+    private double samplingPercentage;
+
+    /**
+     * constructor is responsible of initializing this processor
+     * to default settings
+     */
+    public FixedRateSamplingTelemetryProcessor() {
+        this.samplingPercentage = 100.00;
+        this.includedTypes = new HashSet<Class>();
+        this.excludedTypes = new HashSet<Class>();
+        try {
+            this.allowedTypes = new HashMap<String, Class>() {{
+                put(dependencyTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.RemoteDependencyTelemetry"));
+                put(eventTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.EventTelemetry"));
+                put(exceptionTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.ExceptionTelemetry"));
+                put(pageViewTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.PageViewTelemetry"));
+                put(requestTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.RequestTelemetry"));
+                put(traceTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.TraceTelemetry"));
+            }};
+        } catch (ClassNotFoundException e) {
+            InternalLogger.INSTANCE.trace("Unable to locate telemetry classes");
+        }
+    }
+
+    /**
+     * This method returns a set of classes of excluded types specified by user
+     *
+     * @return
+     */
+    public Set<Class> getExcludedTypes() {
+        return excludedTypes;
+    }
+
+    /**
+     * This method returns a set of classes of included types specified by user
+     *
+     * @return
+     */
+    public Set<Class> getIncludedTypes() {
+        return includedTypes;
+    }
+
+
+    private void setIncludedOrExcludedTypes(String value, Set<Class> typeSet) {
+
+        if (!StringUtils.isNullOrEmpty(value)) {
+            value = value.trim();
+            if (!StringUtils.isNullOrEmpty(value) && allowedTypes.containsKey(value)) {
+                typeSet.add(allowedTypes.get(value));
+            } else {
+                InternalLogger.INSTANCE.error("Item is either not allowed to sample or is empty");
+            }
+        } else {
+            InternalLogger.INSTANCE.error("Empty types cannot be considered");
+        }
+    }
+
+    /**
+     * Gets the sample rate currently set
+     */
+    double getSamplingPercentage() {
+        return samplingPercentage;
+    }
+
+    /**
+     * Sets the user defined sampling percentage
+     *
+     * @param samplingPercentage
+     */
+    public void setSamplingPercentage(String samplingPercentage) {
```

Comment:
why string as an argument?

Reply:
mismatch of `get`-er and `set`-er is not the best practice

Reply:
The string argument is because the Add type takes a parameter string and builds the configuration considering the same. Therefore we need to have a setter which takes the string as an argument. I agree its not the best practice but its how our current configuration is and I might fear that doing some might break other things.